rust/src/dataset/scanner.rs

```diff
@@ -406,18 +399,14 @@ impl Scanner {
         &self,
         filter: Arc<dyn PhysicalExpr>,
         input: Arc<dyn ExecutionPlan>,
-        drop_row_id: bool,
-        ann_schema: Option<Arc<Schema>>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let filter_node = Arc::new(FilterExec::try_new(filter, input)?);
         let output_schema = self.scanner_output_schema()?;
-        Ok(Arc::new(LocalTakeExec::new(
-            filter_node,
+        Ok(Arc::new(TakeExec::try_new(
```

Comment:
why did we need a separate LocalTake before?

Reply:
Was attempted to make it similar to what C++ does, that it takes the same batch from the input BatchRecords, but later found out it is too difficult to implement that in Rust. In the end, `LocalTake` becomes almost identical to `GlobalTake`.